### PR TITLE
chore: Hugh/migrate estimate query cost to v1

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

Taking over from this PR
https://github.com/apache/superset/pull/22910

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by zhangzonghai2021** [comment]
unsubscribe.



***@***.***
 
From: Hugh A. Miles II
Date: 2023-02-28 08:29
To: apache/superset
CC: Subscribed
Subject: [apache/superset] Hugh/migrate estimate query cost to v1 (PR #23226)
SUMMARY
BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
TESTING INSTRUCTIONS
ADDITIONAL INFORMATION
Has associated issue:
Required feature flags:
Changes UI
Includes DB Migration (follow approval process in SIP-59) 
Migration is atomic, supports rollback & is backwards-compatible
Confirm DB migration upgrade and downgrade tested
Runtime estimates and downtime expectations provided
Introduces new feature or API
Removes existing feature or API


You can view, comment on, or merge this pull request online at:
  https://github.com/apache/superset/pull/23226
Commit Summary
f929d56 chore: Migrate /superset/estimate_query_cost/<database_id>/<schema>/ to API v1
a4ec247 Merge branch 'master' of github.com:apache/superset into dm/migrate-estimate_query_cost-to-v1
a7234a8 cleanup
efc3e0e Merge branch 'master' of github.com:apache/superset into dm/migrate-estimate_query_cost-to-v1
c660773 PR comments
File Changes 
(9 files)
M docs/static/resources/openapi.json (201) 
M superset-frontend/src/SqlLab/actions/sqlLab.js (20) 
M superset-frontend/src/SqlLab/reducers/sqlLab.js (2) 
M superset/sqllab/api.py (58) 
A superset/sqllab/commands/estimate.py (106) 
M superset/sqllab/schemas.py (9) 
M superset/views/core.py (1) 
M tests/integration_tests/sql_lab/api_tests.py (67) 
M tests/integration_tests/sql_lab/commands_tests.py (78) 
Patch Links:
https://github.com/apache/superset/pull/23226.patch
https://github.com/apache/superset/pull/23226.diff

File: superset/sqllab/api.py [hunk 1]
```diff
@@ -70,17 +72,71 @@ class SqlLabRestApi(BaseSupersetApi):
...
+                    properties:
+                      result:
+                        type: object
+            400:
+              $ref: '#/components/responses/400'
+            401:
+              $ref: '#/components/responses/401'
+            403:
+              $ref: '#/components/responses/403'
+            404:
```
Comment (dpgaspar) [inline superset/sqllab/api.py:126]: is 404 possible?

File: superset/sqllab/api.py [hunk 2]
```diff
@@ -70,17 +72,69 @@ class SqlLabRestApi(BaseSupersetApi):
...
     }
     openapi_spec_tag = "SQL Lab"
     openapi_spec_component_schemas = (
+        EstimateQueryCostSchema,
         ExecutePayloadSchema,
         QueryExecutionResponseSchema,
     )
 
+    @expose("/estimate/", methods=["POST"])
+    @protect()
```
Comment (dpgaspar) [inline superset/sqllab/api.py:89]: what do you think about adding this endpoint to a `can read on SQLLab` permission name?

File: superset/sqllab/api.py [hunk 3]
```diff
@@ -70,17 +73,70 @@ class SqlLabRestApi(BaseSupersetApi):
...
+    @statsd_metrics
+    @requires_json
+    @permission_name("get")
```
Comment (dpgaspar) [inline superset/sqllab/api.py:93]: sorry, having second thoughts on this, let's remove it and leave the permission has it is. `can estimate query cost on SQLLab`. Also add that permission here: https://github.com/apache/superset/blob/master/superset/security/manager.py#L253

Comment (hughhhh) [reply superset/sqllab/api.py:93]: is that a valid permission? I only see `allows_cost_estimate` as a boolean on the Database model?

Comment (dpgaspar) [reply superset/sqllab/api.py]: it should be can you find `estimate_query_cost` on ab_permission table?

**Review by dpgaspar — APPROVED** [review]

File: docs/static/resources/openapi.json [hunk 4]
```diff
@@ -2627,10 +2628,17 @@
             "type": "string"
           },
           "last_saved_by": {
+<<<<<<< HEAD
```
Comment (dpgaspar) [inline docs/static/resources/openapi.json:2631]: merge conflict here ^^^

**Review by dpgaspar — APPROVED** [review]
LGTM after the docs CI fail is fixed

**Pull request merged**